File: migrate_orders_to_csv.js
const fs = require('fs');

const sql = fs.readFileSync('supabase_seed.sql', 'utf8');
const regex = /INSERT INTO orders \((.*?)\) VALUES \((.*?)\) ON CONFLICT/g;
const headers = ['id', 'address', 'currency', 'customerName', 'customerPhone', 'date', 'items', 'method', 'paymentMethod', 'selectedCompany', 'selectedRegion', 'status', 'timestamp', 'total', 'deliveryCost'];

let rows = [];

let match;
while ((match = regex.exec(sql)) !== null) {
    const colsStr = match[1].replace(/ /g, '');
    const valsStr = match[2];
    
    const parts = [];
    let curr = '';
    let inQuote = false;
    for (let i = 0; i < valsStr.length; i++) {
        const c = valsStr[i];
        if (c === "'") {
            if (i + 1 < valsStr.length && valsStr[i+1] === "'") {
                curr += "'";
                i++;
            } else {
                inQuote = !inQuote;
            }
        } else if (c === ',' && !inQuote) {
            parts.push(curr.trim());
            curr = '';
        } else {
            curr += c;
        }
    }
    parts.push(curr.trim());
    
    const rowDict = {};
    const cols = colsStr.split(',');
    cols.forEach((col, idx) => {
        let val = parts[idx] || '';
        val = val.replace(/\n/g, '\\n').replace(/\r/g, ''); // Fix broken lines
        rowDict[col] = val;
    });
    
    const row = headers.map(h => {
        let v = rowDict[h] || '';
        // Add quotes if comma present
        if (v.includes(',')) return `"${v}"`;
        return v;
    });
    rows.push(row.join(','));
}

fs.writeFileSync('data/orders.csv', headers.join(',') + '\n' + rows.join('\n'));
console.log('Orders migrated.');

File: migrate_orders_to_csv.py
import re
import csv

with open('supabase_seed.sql', 'r', encoding='utf-8') as f:
    sql = f.read()

matches = re.finditer(r"INSERT INTO orders \((.*?)\) VALUES \((.*?)\) ON CONFLICT", sql, re.DOTALL)
headers = ['id', 'address', 'currency', 'customerName', 'customerPhone', 'date', 'items', 'method', 'paymentMethod', 'selectedCompany', 'selectedRegion', 'status', 'timestamp', 'total', 'deliveryCost']

rows = []
for match in matches:
    cols_str = match.group(1).replace(' ', '')
    vals_str = match.group(2)
    
    parts = []
    curr = ''
    in_quote = False
    i = 0
    while i < len(vals_str):
        c = vals_str[i]
        if c == "'":
            if i + 1 < len(vals_str) and vals_str[i+1] == "'":
                curr += "'"
                i += 1
            else:
                in_quote = not in_quote
        elif c == ',' and not in_quote:
            parts.append(curr.strip())
            curr = ''
        else:
            curr += c
        i += 1
    parts.append(curr.strip())
    
    row_dict = {}
    cols = cols_str.split(',')
    for col, val in zip(cols, parts):
        val = val.replace('\n', '\\n').replace('\r', '') # Prevent broken CSV rows
        row_dict[col] = val
    
    row = []
    for h in headers:
        row.append(row_dict.get(h, ''))
    rows.append(row)

with open('data/orders.csv', 'w', newline='', encoding='utf-8') as f:
    writer = csv.writer(f)
    writer.writerow(headers)
    writer.writerows(rows)
print('Orders migrated.')
